### PR TITLE
Add RegisterNode function

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Adds `NodeRegistryAdmin` struct to handle administrative actions on the Nodes smart contract
- Makes it possible to deploy fresh smart contracts in tests, so we can execute against a clean contract state

## More info

We have one blockchain node shared in the test environment, but we are going to want to be able to write tests that run against the same smart contract without interfering with one another.

To handle that we now have a `deploy-ephemeral` script which will create a new instance of the smart contract at a new address. We can call that from inside tests to get a clean contract to test against.

I still have the first deployed contract being loaded from the `_build` folder, which is being used in some older tests. We may want to phase that out and just always work against clean smart contracts.

### CODE REVIEW

**Comment by neekolas** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/143?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#143** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/143?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#142** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/142?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pkg/blockchain/registryAdmin.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+*
+A NodeRegistryAdmin is a struct responsible for calling admin functions on the node registry
+*
+*/
+type NodeRegistryAdmin struct {
+	client   *ethclient.Client
+	signer   TransactionSigner
+	contract *abis.Nodes
+	logger   *zap.Logger
+}
```
Comment (richardhuaaa) [inline pkg/blockchain/registryAdmin.go:28]: Is this used in production code, or only in unit tests?

**Review by richardhuaaa — APPROVED** [review]

File: pkg/blockchain/registryAdmin.go (hunk 1 above)
Comment (neekolas) [reply pkg/blockchain/registryAdmin.go:28]: Great question. 

Unit tests right now, but I think we might want a CLI command to do admin actions at least in the testnet phase. So, not purely a test util.
